src/libraries/Common/src/System/Net/CookieParser.cs

```diff
@@ -615,17 +616,17 @@ internal Cookie Get()
                                     if (!commentSet)
                                     {
                                         commentSet = true;
-                                        cookie.Comment = _tokenizer.Value;
+                                        cookie!.Comment = _tokenizer.Value;
```

Comment:
What prevents cookie from being null here?

Reply:
_tokenizer.Next(cookie == null, true) returns NameValuePair, Attribute or terminator. For first two we call new Cookie(), for last we skip all cases in the switch statement.